File: #03/Categoria 04/q15 - swap/swap.js
swap = (arr, a , b) => 
(arr[a] && arr[b]) ? 
    arr.slice(0, a).concat(arr[b]).concat(arr.slice(a+1, b)).concat(arr[a]).concat(arr.slice(b+1, arr.length)) 
: arr

console.log(swap([], 0, 5));
console.log(swap([1], 0, 3));
console.log(swap([1,3,4], 1, 2));
console.log(swap([1,2,3,4,5,6], 2, 5));
console.log(swap([5,6,7,8,9], 0, 3));

File: #03/Categoria 04/q15 - swap/swap.hs
slice :: [a] -> Int -> Int -> [a]
slice xs a b = take (b - a + 1) (drop a xs)

checkIndex :: Foldable t => t a -> Int -> Bool
checkIndex xs a =
    (length xs - 1) >= a

swap :: [Int] -> Int -> Int -> [Int]
swap xs a b =
    if checkIndex xs a && checkIndex xs b
        then slice xs 0 (a-1) ++ [xs !! b] ++ slice xs (a+1) (b-1) ++ [xs !! a] ++ slice xs (b+1) (length xs)
    else xs

main :: IO ()
main = do
    a <- readLn :: IO [Int]
    b <- readLn :: IO Int
    c <- readLn :: IO Int
    print $ swap a b c